beacon-chain/core/light-client/lightclient.go

```diff
@@ -592,6 +592,10 @@ func HasFinality(update interfaces.LightClientUpdate) (bool, error) {
 }
 
 func IsBetterUpdate(newUpdate, oldUpdate interfaces.LightClientUpdate) (bool, error) {
+	if oldUpdate == nil {
```

Comment:
This will never be true because the object is an interface. You need an `IsNil()` function just like there is one for finality and optimistic interfaces